R/hello.R

```diff
@@ -2,21 +2,26 @@
 #'
 #' Reads a message with a name and returns a message greeting the name.
 #' @references \url{https://github.com/grpc/grpc/tree/master/examples/cpp/helloworld}
-#' @importFrom RProtoBuf readProtoFiles read new serialize
+#' @importFrom RProtoBuf readProtoFiles read new serialize P
 helloworld <- function() {
 
     ## reading the service definitions
-    readProtoFiles(system.file('examples/helloworld.proto', package = 'grpc'))
-
+    spec <- system.file('examples/helloworld.proto', package = 'grpc')
+    
+    impl <- read_services(spec)
+  
     ## defining service handlers
-    handler <- list(
-        '/helloworld.Greeter/SayHello' = function(x) {
-            request  <- read(helloworld.HelloRequest, x)
-            response <- new(helloworld.HelloReply, message = paste('Hello, ', request$name))
-            serialize(response, NULL)
-        })
+    SayHello <- grep('SayHello', names(impl))
+    
+    #hmmm slightly gross, write a helper
+    impl[[SayHello]]$f <- function(request){
```

Comment:
It's very nice that we don't have to copy/paste the boilerplate from the proto file manually, but the previous version was still a lot more intuitive, so :+1:  for a helper :)

Reply:
PS let me know if you need any help with that -- I might be able to help with that.